.github/pull_request_template.md

```diff
@@ -0,0 +1,16 @@
+### Description
```

Comment:
I might be missing something but why are we introducing a PR template in each client repo? We removed them in PR #114  to follow the standard templates from `.github` repo

Reply:
@VachaShah we now have a changelog that needs to be updated as part of every PR. I thought it'd be best to track that and a bunch of other requirements like DCO in the PR template. We've been seeing a lot more PRs missing asks around dco, tests etc.

Reply:
+1 to @VachaShah comment: we have the option of a) create templates everywhere b) not put anything about CHANGELOG in the generic template (DCO is everywhere)

I think since CI will fail without CHANGELOG or DCO the checkbox is maybe unnecessary?